core/users.js

```diff
@@ -0,0 +1,46 @@
+// import { CastError } from 'mongoose';
+import jwt from 'jsonwebtoken';
+import encryption from 'utils/encryption';
+import { UserNotFoundError, PasswordNotMatchError } from 'common/error';
+import User from 'db/models/User';
+
+const secretKey = process.env.SECRET_KEY;
+
+/**
+ *
+ * Find an user that matches given userId
+ * @param {String} userId
+ * @returns {Object} response data: User
+ */
+async function findById(userId) {
+  let data = {};
+  const user = await User.findOne(
+    { _id: userId },
+    { _id: 0, password_hashed: 0, password_salt: 0 },
+  );
+
+  if (!user) {
+    const err = new UserNotFoundError();
+    throw err;
+  }
+  data = user;
+  return data;
+}
+
+async function checkLogin(usr, pwd) {
+  let data = {};
+  const user = await User.findOne({ username: usr });
+  if (!user) {
+    const err = new UserNotFoundError({ code: 40402, message: 'Invalid Username' });
+    throw err;
+  }
+  if (encryption.isEqual(pwd, user.get('password_hashed'))) {
```

Comment:
`user.get('password_hashed')`? The model do not have method to get or set for each field?
We should not use string (constant) to get a property